localstack/services/apigateway/integration.py

```diff
@@ -324,187 +287,107 @@ def invoke(self, invocation_context: ApiInvocationContext) -> Response:
+                    ),
+                )
 
-    def urlDecode(self, s):
-        return unquote_plus(s)
+            result = json_safe(result)
+            response = requests_response(content=result)
 
-    def escapeJavaScript(self, s):
-        try:
-            return json.dumps(json.loads(s))
-        except Exception:
-            primitive_types = (str, int, bool, float, type(None))
-            s = s if isinstance(s, primitive_types) else str(s)
-        if str(s).strip() in {"true", "false"}:
-            s = bool(s)
-        elif s not in [True, False] and is_number(s):
-            s = to_number(s)
-        return json.dumps(s)
-
-
-class VelocityInput:
-    """
-    Simple class to mimic the behavior of variable '$input' in AWS API Gateway integration
-    velocity templates.
-    See: http://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-mapping-template-reference.html
-    """
-
-    def __init__(self, body, params):
-        self.parameters = params or {}
-        self.value = body
-
-    def path(self, path):
-        if not self.value:
-            return {}
-        value = self.value if isinstance(self.value, dict) else json.loads(self.value)
-        return extract_jsonpath(value, path)
-
-    def json(self, path):
-        path = path or "$"
-        matching = self.path(path)
-        if isinstance(matching, (list, dict)):
-            matching = json_safe(matching)
-        return json.dumps(matching)
-
-    @property
-    def body(self):
-        return self.value
-
-    def params(self, name=None):
-        if not name:
-            return self.parameters
-        for k in ["path", "querystring", "header"]:
-            if val := self.parameters.get(k).get(name):
-                return val
-        return ""
-
-    def __getattr__(self, name):
-        return self.value.get(name)
-
-    def __repr__(self):
-        return "$input"
-
-
-class ApiGatewayVtlTemplate(VtlTemplate):
-    """Util class for rendering VTL templates with API Gateway specific extensions"""
-
-    def prepare_namespace(self, variables) -> Dict[str, Any]:
-        namespace = super().prepare_namespace(variables)
-        if stage_var := variables.get("stage_variables") or {}:
-            namespace["stageVariables"] = stage_var
-        input_var = variables.get("input") or {}
-        variables = {
-            "input": VelocityInput(input_var.get("body"), input_var.get("params")),
-            "util": VelocityUtilApiGateway(),
-        }
-        namespace.update(variables)
-        return namespace
-
-
-class Templates:
-    __slots__ = ["vtl"]
+        # apply response templates
+        invocation_context.response = response
+        response._content = self.response_templates.render(invocation_context)
+        return response
 
-    def __init__(self):
-        self.vtl = ApiGatewayVtlTemplate()
 
-    def render(self, api_context: ApiInvocationContext) -> Union[bytes, str]:
-        pass
+class IntegrationSubtypes:
```

Comment:
Since integration subtypes are a v2 concept, I'd suggest to move them upstream into -ext, in order not to mix the concepts here. Thoughts?

Reply:
Yep, it does make sense 👍  at the time (still true today), the mapping `integration_subtypes` is dependent on the type `StepFunctionIntegration`, which is declared in the community, thus the definition here. I'll move the code to upstream!